plugins/rich-editor/src/scripts/components/EditorEmojiPicker.jsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * @author Stéphane (slafleche) LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+import React from "react";
+import * as PropTypes from "prop-types";
+import Quill from "quill/quill";
+import EditorEmojiMenu from "../components/EditorEmojiMenu";
+import { t } from "@core/utility";
+
+
+export default class EditorEmojiPicker extends React.Component {
+
+    /** @type {number} */
+    static count;
+
+    static propTypes = {
+        quill: PropTypes.instanceOf(Quill).isRequired,
+    };
+
+    /** @type {Quill} */
+    quill;
+
+    /**
+     * @type {Object}
+     * @property {RangeStatic} - The current quill selected text range..
+     */
+    state;
+
+    /**
+     * @inheritDoc
+     */
+    constructor(props) {
+        super(props);
+
+        if (!this.count) {
+            this.count = 1;
```

Comment:
This incrementing doesn't seem to actually work. AFAIK you can't access static members through the this property. I think you need a factory pattern to implement this in javascript. Do we even need these ID's? What are they used for?

Reply:
It's for accessibility. Some aria attributes require an ID.